src/cluster/replication.cc

```diff
@@ -603,6 +667,7 @@ ReplicationThread::CBState ReplicationThread::incrementBatchLoopCB(bufferevent *
                 util::StringToHex(batch.Data()));
           return CBState::RESTART;
         }
+        data_written = true;
```

Comment:
I'm not sure why do we need to call `sendReplConfAck` in other places? I think it would be good to send the ack after applying the write batch only. 

What I mean is to remove `data_written` and `sendReplConfAck` in other places, and just call `sendReplConfAck` here. Can help to correct me if I missed anything.

Reply:
It is possible in one `incrementBatchLoopCB` call, there are several `rocksdb::WriteBatch` can be constructed from existing data in bufferevent (in my local test env, there can can anything from 3 to 22 write batches in a single call). It is unnecessary and potentially wasteful to send an ack for each of them. Instead, we can just write all the batches to RocksDB (assuming write to RocksDB is fast) and send ack once.

One thing I would potentially do in a following PR is to make this behavior configurable. e.g. user may specify ack on every send / ack on every n send or keep the current behavior. But I would want to benchmark the throughput first. For now, I think it is a reasonable default behavior.